src/librustc_metadata/creader.rs

```diff
@@ -853,6 +853,33 @@ impl<'a> CrateLoader<'a> {
         }
     }
 
+    fn inject_profiler_runtime(&mut self) {
+        if self.sess.opts.debugging_opts.profile {
+            let mut uses_std = false;
+            self.cstore.iter_crate_data(|_, data| {
```

Comment:
@alexcrichton iirc, because the sanitizers only work with `alloc_system` (they depend on `memcpy` and co.), use `libc` functions like `write` to print results to the console and pretty much depend on the Linux kernel so you need an OS to use them and that's guaranteed to be there when you link to `std`.

But since `-Z sanitizer` is restricted to only work with the `x86_64-unknown-linux-gnu` target, this check may not be that necessary. Still `no_std` + `x86_64-unknown-linux-gnu` will not work unless you link to `libc`.